src/react-components/home/usePublicRooms.js

```diff
@@ -0,0 +1,14 @@
+import { useCallback, useContext } from "react";
+import { usePaginatedAPI } from "./usePaginatedAPI";
+import { fetchReticulumAuthenticated } from "../../utils/phoenix-utils";
+import { AuthContext } from "../auth/AuthContext";
+
+export function usePublicRooms() {
+  const auth = useContext(AuthContext); // Re-render when you log in/out.
+  const getMoreRooms = useCallback(
+    cursor => fetchReticulumAuthenticated(`/api/v1/media/search?source=rooms&filter=public&cursor=${cursor}`),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [auth.isSignedIn]
```

Comment:
Can't public rooms be enabled based on a permission? `getMoreRooms` needs to change so that `usePaginatedAPI` fetches again. That's how the API works.